serve/pull.go

```diff
@@ -9,91 +9,88 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"log"
 	"net/http"
 	"strconv"
 	"strings"
 
-	"github.com/attic-labs/noms/go/datas"
 	"github.com/attic-labs/noms/go/hash"
 	"github.com/attic-labs/noms/go/types"
+	zl "github.com/rs/zerolog"
 
 	"roci.dev/diff-server/db"
 	"roci.dev/diff-server/kv"
 	servetypes "roci.dev/diff-server/serve/types"
 	nomsjson "roci.dev/diff-server/util/noms/json"
 )
 
-func (s *Service) pull(rw http.ResponseWriter, req *http.Request) {
+func (s *Service) pull(rw http.ResponseWriter, req *http.Request, l zl.Logger) {
 	if req.Method != "POST" {
-		unsupportedMethodError(rw, req.Method)
+		unsupportedMethodError(rw, req.Method, l)
 		return
 	}
 
 	body := bytes.Buffer{}
 	_, err := io.Copy(&body, req.Body)
 	if err != nil {
-		serverError(rw, fmt.Errorf("could not read body: %w", err))
+		serverError(rw, fmt.Errorf("could not read body: %w", err), l)
 		return
 	}
 
 	var preq servetypes.PullRequest
 	err = json.Unmarshal(body.Bytes(), &preq)
 	if err != nil {
-		serverError(rw, fmt.Errorf("could not unmarshal body to json: %w", err))
+		serverError(rw, fmt.Errorf("could not unmarshal body to json: %w", err), l)
 		return
 	}
 
 	// TODO auth
 	accountName := req.Header.Get("Authorization")
 	if accountName == "" {
-		clientError(rw, http.StatusBadRequest, "Missing Authorization header")
+		clientError(rw, http.StatusBadRequest, "Missing Authorization header", l)
 		return
 	}
 	acct, ok := lookupAccount(accountName, s.accounts)
 	if !ok {
-		clientError(rw, http.StatusBadRequest, fmt.Sprintf("Unknown account: %s", accountName))
+		clientError(rw, http.StatusBadRequest, fmt.Sprintf("Unknown account: %s", accountName), l)
 		return
 	}
 
 	if preq.ClientID == "" {
-		clientError(rw, http.StatusBadRequest, "Missing clientID")
+		clientError(rw, http.StatusBadRequest, "Missing clientID", l)
 		return
 	}
 
 	db, err := s.GetDB(accountName, preq.ClientID)
 	if err != nil {
-		serverError(rw, err)
+		serverError(rw, err, l)
 		return
 	}
 
-	logPayload(req, body.Bytes(), db.Noms())
-
 	fromHash, ok := hash.MaybeParse(preq.BaseStateID)
 	if preq.BaseStateID != "" && !ok {
-		clientError(rw, http.StatusBadRequest, "Invalid baseStateID")
+		clientError(rw, http.StatusBadRequest, "Invalid baseStateID", l)
 		return
 	}
 	fromChecksum, err := kv.ChecksumFromString(preq.Checksum)
 	if err != nil {
-		clientError(rw, http.StatusBadRequest, "Invalid checksum")
+		clientError(rw, http.StatusBadRequest, "Invalid checksum", l)
 		return
 	}
 
 	clientViewURL := acct.ClientViewURL
 	if s.overridClientViewURL != "" {
-		log.Printf("WARNING: overriding all client view URLs with %s", s.overridClientViewURL)
+		l.Debug().Msgf("WARNING: overriding all client view URLs with %s", s.overridClientViewURL)
```

Comment:
let's axe the WARNING in the string. also i think this should probably be info?

Reply:
The thing is that this message is pretty annoying right now because it shows up on every pull. It makes it seems like something's wrong, when there's nothing wrong, it's doing exactly what it is supposed to do. Maybe this should actually be printed once at startup?

Reply:
that sounds way better